devices/Bq2579x/Bq2579x.cs

```diff
@@ -0,0 +1,939 @@
+
+        /// <summary>
+        /// Gets die temperature.
+        /// </summary>
+        /// <remarks>Range: -40�C -150�C.</remarks>
+        public Temperature DieTemperature => GetDieTemperature();
+
+        /// <summary>
+        /// Gets or sets the watchdog timer setting (in milliseconds).
+        /// </summary>
+        public WatchdogSetting WatchdogTimerSetting { get => GetWatchdogTimerSetting(); set => SetWatchdogTimerSetting(value); }
+
+        /// <summary>
+        /// Gets or sets Thermal regulation threshold.
+        /// </summary>
+        [Property]
+        public ThermalRegulationThreshold ThermalRegulationThreshold { get => GetThermalRegulationThreshold(); set => SetThermalRegulationThreshold(value); }
+
+        /// <summary>
+        /// Gets or sets Thermal shutdown threshold.
+        /// </summary>
+        [Property]
+        public ThermalShutdownThreshold ThermalShutdownThreshold { get => GetThermalShutdownThreshold(); set => SetThermalShutdownThreshold(value); }
+
+        /// <summary>
+        /// Gets Power Good Status.
+        /// </summary>
+        /// <value><see langword="true"/> if power is good, otherwise <see langword="false"/>.</value>
+        [Property]
+        public bool PowerGood => GetChargerStatus0().HasFlag(ChargerStatus0.PowerGood);
+
+        /// <summary>
+        /// Gets state of VAC2 inserted.
+        /// </summary>
+        /// <value><see langword="true"/> if VAC2 is inserted, otherwise <see langword="false"/>.</value>
+        [Property]
+        public bool VAC2Present => GetChargerStatus0().HasFlag(ChargerStatus0.Vac2Inserted);
+
+        /// <summary>
+        /// Gets state of VBUS inserted.
+        /// </summary>
+        /// <value><see langword="true"/> if VBUS is inserted, otherwise <see langword="false"/>.</value>
+        [Property]
+        public bool VBUSPresent => GetChargerStatus0().HasFlag(ChargerStatus0.VbusPresent);
+
+        /// <summary>
+        /// Gets state of VAC1 inserted.
+        /// </summary>
+        /// <value><see langword="true"/> if VAC1 is inserted, otherwise <see langword="false"/>.</value>
+        [Property]
+        public bool VAC1Present => GetChargerStatus0().HasFlag(ChargerStatus0.Vac1Inserted);
+
+        /// <summary>
+        /// Gets state IC thermal regulation.
+        /// </summary>
+        /// <value><see langword="true"/> if device is in thermal regulation, <see langword="false"/> for normal operation.</value>
+        [Property]
+        public bool IsInThermalRegulation => GetChargerStatus2().HasFlag(ChargerStatus2.ThermalRegulation);
+
+        /// <summary>
+        /// Gets D+/D- detection status bits.
+        /// </summary>
+        /// <value><see langword="true"/> if D+/D- detection is ongoing, <see langword="false"/>  D+/D- detection is NOT started yet, or the detection is done.</value>
+        [Property]
+        public bool DpDmDetectionOngoing => GetChargerStatus2().HasFlag(ChargerStatus2.DpDmStatus);
```

Comment:
This one is fine without the `Is` because of the `Ongoing` which implicitly means `Is`